pvlib/pvsystem.py

```diff
@@ -1064,6 +1064,144 @@ def physicaliam(aoi, n=1.526, K=4., L=0.002):
     return iam
 
 
+def iam_martin_ruiz(θ:'degrees', a_r=0.16):
+    '''
+    Determine the incidence angle modifier using the Martin
+    and Ruiz incident angle model
+
+    iam_martin_ruiz calculates the incidence angle modifier (angular
+    factor) as described by Martin and Ruiz in [1]. The information
+    required is the incident angle (theta) and the angular losses
+    coefficient (ar). Please note that [1] has a corrigendum which makes the
+    document much simpler to understand.
+
+    The incident angle modifier is defined as
+    [1-exp(-cos(theta/ar))] / [1-exp(-1/ar)], which is
+    presented as AL(alpha) = 1 - IAM in equation 4 of [1]. Thus IAM is
+    equal to 1 at theta = 0, and equal to 0 at theta = 90. IAM is a
+    column vector with the same number of elements as the largest input
+    vector.
+
+    Parameters
+    ----------
+    θ : numeric, degrees
+        The angle of incidence between the module normal vector and the
+        sun-beam vector in degrees. Theta must be a numeric scalar or vector.
+        iam is 0 where |θ| > 90
+
+    a_r : numeric
+        The angular losses coefficient described in equation 3 of [1].
+        This is an empirical dimensionless parameter. Values of ar are
+        generally on the order of 0.08 to 0.25 for flat-plate PV modules. a_r
+        must be a positive numeric scalar.
+
+    Returns
+    -------
+    iam : numeric
+        The incident angle modifier(s)
+
+    References
+    ----------
+    [1] N. Martin and J. M. Ruiz, "Calculation of the PV modules angular
+    losses under field conditions by means of an analytical model", Solar
+    Energy Materials & Solar Cells, vol. 70, pp. 25-38, 2001.
+
+    [2] N. Martin and J. M. Ruiz, "Corrigendum to 'Calculation of the PV
+    modules angular losses under field conditions by means of an
+    analytical model'", Solar Energy Materials & Solar Cells, vol. 110,
+    pp. 154, 2013.
+
+    See Also
+    --------
+    physicaliam
+    ashraeiam
+    iam_interp
+    '''
+    # Contributed by Anton Driesse (@adriesse), PV Performance Labs. July, 2019
+
+    θ = np.asanyarray(θ)
+    a_r = np.asanyarray(a_r)
+
+    if not np.all(np.positive(a_r)):
+        raise RuntimeError("The parameter 'a_r' cannot be zero or negative.")
+
+    θ = np.clip(θ, -90, 90)
+    iam = (1 - np.exp(-cosd(θ)/ a_r)) / (1 - np.exp(-1 / a_r))
+
+    return iam
+
+
+def iam_interp(θ:'degrees', θ_ref, iam_ref, method='linear', normalize=True):
```

Comment:
Not necessarily objecting, but wondering about the motivation for this function. It's a very thin wrapper over scipy and it's not clear to me that pvlib provides value over users directly calling the scipy function. (I vaguely recall having this discussion previously... sorry if I've forgotten it)

Certainly would be a good example or stackoverflow question/answer if not a function here.

Reply:
I think it would be helpful to offer a function that is recognized as implementing the interpolation model for IAM.  The computation isn't the issue, rather, it's having the model choices include interpolation as an option.

Reply:
I do recall your earlier comments @wholmgren .  I think the function includes some application-specific aspects that are useful and help to standardize the procedure.